sky/serve/replica_managers.py

```diff
@@ -440,6 +492,18 @@ def probe(
                         f'{common_utils.format_exception(e)}.')
         return self, False, probe_time
 
+    def __setstate__(self, state):
+        """Set state from pickled state, for backward compatibility."""
+        version = state.pop('_version', None)
+        # Handle old version(s) here.
+        if version is None:
+            version = -1
+
+        if version < 0:
+            self.is_spot = False
```

Comment:
humm is that true? how about old service with spot deployment?

Reply:
it is fine as it will be handled only with `RequestRateAutoscaler`, treated just like the on-demand replica.

Reply:
add a comment for this?